changelog.d/15748.removal

```diff
@@ -0,0 +1 @@
+Remove experimental [MSC2716](https://github.com/matrix-org/matrix-spec-proposals/pull/2716) implementation to incrementally import history into existing rooms.
```

Comment:
It looks like the current CI failures are [from `develop`](https://github.com/matrix-org/synapse/actions/runs/5279369630/jobs/9550008431) :x: